### PR TITLE
Processing data in models

### PR DESCRIPTION
**Pull request by devMahmoud**
## In this milestone I did:
* Create model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
* Set up associations between models.
* Create custom methods

        A method that returns the 3 most recent posts for a given user.
        A method that updates the posts counter for a user.
        A method that returns the 5 most recent comments for a given post.
        A method that updates the comments counter for a post.
        A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :comments
+  has_many :posts
+  has_many :likes
```
Comment (ShoiraTa) [inline app/models/user.rb]: - [x] You have done a great job implementing relationships, considering that using user as author is very confusing. The only thing you missed here, is adding foreign key to user like: 
```ruby
has_many :posts, foreign_key: 'author_id'
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'
```

Without foreign key you cannot connect user with posts, you will be getting an error: 
Screenshot 2022-08-11 at 20 14 38

after adding foreign keys everything works as expected 👍🏻

**Review by ShoiraTa — CHANGES_REQUESTED** [review]
Hi @devMahmoud ,

:thumbsup: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

**First let me to point out:**

- Project is setup correctly ✔️ 
- Required methods are added  in models ✔️ 
- You passed all the Linters! ✔️ 
- Professional pull request! ✔️ 

Keep on coding, good job so far! :clap:

There are some issues that you still need to work on to go to the next project but you are almost there!



 ## Required Changes  :recycle: 

_Check the comments under the review._ 

***
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear by mentioning my Github username.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

* * *
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (peterrobert) [reply app/models/user.rb]: This change has been done well

**Review by peterrobert — APPROVED** [review]
Hi @devMahmoud, 

### first let me point out:

- [ ] Great work with the models and the relationships 👏🏿 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**